core/src/main/java/org/lflang/analyses/uclid/UclidGenerator.java

```diff
@@ -0,0 +1,1733 @@
+            + (diagram.tail.index - diagram.loopNode.index + 1) * loopIterations;
+
+        An overflow-safe version of the line above
+        */
+        int t0 = Math.addExact(diagram.loopNode.index, 1);
+        int t1 = Math.subtractExact(diagram.tail.index, diagram.loopNode.index);
+        int t2 = Math.addExact(t1, 1);
+        int t3 = Math.multiplyExact(t2, loopIterations);
+        this.CT = Math.addExact(t0, t3);
+
+      } else {
+        // Get the number of events before the loop starts.
+        // This stops right before the loopNode is encountered.
+        StateSpaceNode node = diagram.head;
+        int numReactionInvocationsBeforeLoop = 0;
+        while (node != diagram.loopNode) {
+          numReactionInvocationsBeforeLoop += node.reactionsInvoked.size();
+          node = diagram.getDownstreamNode(node);
+        }
+        // Account for the loop node in numReactionInvocationsBeforeLoop.
+        numReactionInvocationsBeforeLoop += node.reactionsInvoked.size();
+
+        // Count the events from the loop node until
+        // loop node is reached again.
+        int numReactionInvocationsInsideLoop = 0;
+        do {
+          node = diagram.getDownstreamNode(node);
+          numReactionInvocationsInsideLoop += node.reactionsInvoked.size();
+        } while (node != diagram.loopNode);
+
+        /*
+        CT = steps required for the non-periodic part
+             + steps required for the periodic part
+
+        this.CT = numReactionInvocationsBeforeLoop
+            + numReactionInvocationsInsideLoop * loopIterations;
+
+        An overflow-safe version of the line above
+        */
+        // System.out.println("DEBUG: numReactionInvocationsBeforeLoop: " +
+        // numReactionInvocationsBeforeLoop);
+        // System.out.println("DEBUG: numReactionInvocationsInsideLoop: " +
+        // numReactionInvocationsInsideLoop);
+        // System.out.println("DEBUG: loopIterations: " + loopIterations);
+        int t0 = Math.multiplyExact(numReactionInvocationsInsideLoop, loopIterations);
+        this.CT = Math.addExact(numReactionInvocationsBeforeLoop, t0);
+      }
+    }
+  }
+
+  /** Process an MTL property. */
+  private void processMTLSpec() {
+    MTLLexer lexer = new MTLLexer(CharStreams.fromString(this.spec));
+    CommonTokenStream tokens = new CommonTokenStream(lexer);
+    MTLParser parser = new MTLParser(tokens);
+    MtlContext mtlCtx = parser.mtl();
+    MTLVisitor visitor = new MTLVisitor(this.tactic);
+
+    // The visitor transpiles the MTL into a Uclid axiom.
+    this.FOLSpec = visitor.visitMtl(mtlCtx, "i", 0, "0", 0);
+    this.horizon = visitor.getHorizon();
+  }
+
+  /////////////////////////////////////////////////
+  //// Functions from generatorBase
+
+  @Override
+  public Target getTarget() {
+    return Target.C; // Works with a C subset.
+  }
+
+  @Override
+  public TargetTypes getTargetTypes() {
+    throw new UnsupportedOperationException("TODO: auto-generated method stub");
```

Comment:
It might be wise to provide a short explanation of why it doesn't make sense to implement `getTargetTypes` instead of keeping the text `"TODO: auto-generated method stub"`.